src/main/java/org/opentestsystem/delivery/logging/LoggingInterceptor.java

```diff
@@ -0,0 +1,56 @@
+package org.opentestsystem.delivery.logging;
+
+import com.google.common.base.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.opentestsystem.delivery.logging.EventLogger.Checkpoint.ENTER;
+import static org.opentestsystem.delivery.logging.EventLogger.Checkpoint.EXIT;
+
+/**
+ * Logs all http requests that are handled by methods with the {@link org.springframework.web.bind.annotation.RequestMapping @RequestMapping} annotation.
+ *
+ * Common http fields such as http session and request parameters are logged.
+ * Custom fields can be added by implementing an EventParserFactory subclass and
+ * injecting it via Spring configuration.
+ */
+public class LoggingInterceptor extends HandlerInterceptorAdapter {
+  public static final String EVENT_INFO = "event-info";
+  public static final String LOGGER_NAME = "event-logger";
+  private final EventLogger logger;
+  private final EventParserFactory factory;
+
+  @Autowired
+  public LoggingInterceptor(final EventParserFactory factory, final EventLogger logger) {
+    this.factory = factory;
+    this.logger = logger;
+  }
+
+  @Override
+  public boolean preHandle(HttpServletRequest request,
+                           HttpServletResponse response, Object handler) throws Exception {
+    final EventParser parser = factory.get(request);
+    final Optional<EventInfo> eventInfo = parser.parsePreHandle(request, handler, logger);
+    if (eventInfo.isPresent()) {
+      logger.info(eventInfo.get().withCheckpoint(ENTER.name()));
+      // TODO: make sure these are needed - logger should be injected, and contains a field stash.
+      request.setAttribute(EVENT_INFO, eventInfo.get());
+      request.setAttribute(LOGGER_NAME, logger);
+    }
```

Comment:
@mtrupkin , Re: the TODO note I made above - do we need the logger and eventInfo put into the request as attributes? I haven't used this method of access yet in student - did you need it in proctor? This might be leftover from our old plan. Either way let's address then remove the TODO note.